crates/viewer/re_blueprint_tree/src/blueprint_tree.rs

```diff
@@ -788,25 +792,30 @@ impl BlueprintTree {
                 usize::MAX,
             );
 
-            self.handle_drop_target(viewport, ui, dragged_item_id, &drop_target);
+            self.handle_contents_drop_target(viewport, ui, dragged_contents, &drop_target);
         }
     }
 
-    fn handle_drop_target(
+    fn handle_contents_drop_target(
         &mut self,
         viewport: &ViewportBlueprint,
         ui: &Ui,
-        dragged_item_id: Contents,
+        dragged_contents: &[Contents],
         drop_target: &DropTarget<Contents>,
     ) {
-        // We cannot allow the target location to be "inside" the dragged item, because that would amount moving
-        // myself inside of me.
-        if let Contents::Container(dragged_container_id) = &dragged_item_id {
-            if viewport
-                .is_contents_in_container(&drop_target.target_parent_id, dragged_container_id)
-            {
-                return;
+        // We cannot allow the target location to be "inside" any of the dragged items, because that
+        // would amount to moving myself inside of me.
+        if dragged_contents.iter().any(|dragged_contents| {
+            if let Contents::Container(dragged_container_id) = dragged_contents {
+                if viewport
+                    .is_contents_in_container(&drop_target.target_parent_id, dragged_container_id)
+                {
+                    return true;
+                }
             }
+            false
+        }) {
```

Comment:
this would be more readable if you broke out the boolean and/or lambda to a named variable:

``` rs
let parent_contains_dragged_content = |dragged_contents| { … };
if dragged_contents.iter().any(parent_contains_dragged_content) {
```